nodebuilder/tests/tastora/framework.go

```diff
@@ -0,0 +1,542 @@
+	require.NoError(f.t, err, "failed to start full node")
+	return fullNode
+}
+
+// startLightNode initializes and starts a light node.
+func (f *Framework) startLightNode(ctx context.Context, fullNode tastoratypes.DANode, chain tastoratypes.Chain) tastoratypes.DANode {
+	genesisHash := f.getGenesisHash(ctx, chain)
+
+	p2pInfo, err := fullNode.GetP2PInfo(ctx)
+	require.NoError(f.t, err, "failed to get full node p2p info")
+
+	p2pAddr, err := p2pInfo.GetP2PAddress()
+	require.NoError(f.t, err, "failed to get full node p2p address")
+
+	lightNode := f.daNetwork.GetLightNodes()[0]
+	err = lightNode.Start(ctx,
+		tastoratypes.WithChainID(testChainID),
+		tastoratypes.WithAdditionalStartArguments("--p2p.network", testChainID, "--rpc.addr", "0.0.0.0"),
+		tastoratypes.WithEnvironmentVariables(
+			map[string]string{
+				"CELESTIA_CUSTOM": tastoratypes.BuildCelestiaCustomEnvVar(testChainID, genesisHash, p2pAddr),
+				"P2P_NETWORK":     testChainID,
+			},
+		),
+	)
+	require.NoError(f.t, err, "failed to start light node")
+	return lightNode
+}
+
+// getGenesisHash returns the genesis hash of the chain.
+func (f *Framework) getGenesisHash(ctx context.Context, chain tastoratypes.Chain) string {
+	node := chain.GetNodes()[0]
+	c, err := node.GetRPCClient()
+	require.NoError(f.t, err, "failed to get node client")
+
+	first := int64(1)
+	block, err := c.Block(ctx, &first)
+	require.NoError(f.t, err, "failed to get block")
+
+	genesisHash := block.Block.Header.Hash().String()
+	require.NotEmpty(f.t, genesisHash, "genesis hash is empty")
+	return genesisHash
+}
+
+// cleanup frees up all resources.
+func (f *Framework) cleanup() {
+	f.logger.Info("Cleaning up Tastora framework")
+	// Cleanup is handled by the Tastora framework automatically
+}
+
+// appOverrides modifies the "app.toml" configuration.
+func appOverrides() toml.Toml {
+	appTomlOverride := make(toml.Toml)
+	txIndexConfig := make(toml.Toml)
+	txIndexConfig["indexer"] = "kv"
+	appTomlOverride["tx-index"] = txIndexConfig
+	return appTomlOverride
+}
+
+// configOverrides modifies the "config.toml" configuration.
+func configOverrides() toml.Toml {
+	overrides := make(toml.Toml)
+	txIndexConfig := make(toml.Toml)
+	txIndexConfig["indexer"] = "kv"
+	overrides["tx_index"] = txIndexConfig
+	return overrides
+}
+
+// getCelestiaTag returns the Celestia app image tag.
+func getCelestiaTag() string {
+	if tag := os.Getenv("CELESTIA_TAG"); tag != "" {
+		return tag
+	}
+	return defaultCelestiaTag
+}
+
+// getNodeTag returns the Celestia node image tag.
+func getNodeTag() string {
+	if tag := os.Getenv("CELESTIA_NODE_TAG"); tag != "" {
+		return tag
+	}
+	return defaultNodeTag
+}
+
+// getNodeImage returns the Celestia node image.
+func getNodeImage() string {
+	if tag := os.Getenv("CELESTIA_NODE_IMAGE"); tag != "" {
+		return tag
+	}
+	return nodeImage
+}
+
+// GenerateTestBlobs generates test blobs for blob operations
+func (f *Framework) GenerateTestBlobs(count int, dataSize int) ([]*blob.Blob, error) {
```

Comment:
Shouldn't be a method, and part of this file. It doesn't use any fields on the Framework and is quite irrelevant to its responsibility. I would move it to test utilities next to the blob pkg or at least a separate utilities file. Same as other generator methods.